project/CONTRIBUTING.md

```diff
@@ -422,6 +423,78 @@ Thank them for contributing. Encourage them if you feel this is going to be long
 In short, try to make them want to contribute again. If they are stuck, try to provide them with
 code yourself, or ping someone who can help.
 
+## Drafting a new release
+
+### Minor releases and patch releases
+
+Releasing software is the last step to getting your bugfixes or new features to your user base,
+and should be done regularly, so that users are not tempted to use development branches.
+To know what you are going to release on branch 42.x, given that the last release on this branch is 42.3.1,
+go to `https://github.com/sonata-project/SonataCoreBundle/compare/42.3.1...42.x`.
+You should see a list of commits, some of which should contain links to pull requests.
+
+#### Determining the next release number
+
+If any of those pull requests is labeled `minor`, then the next release should be a minor release (42.4.0).
+Otherwise, if there are any pull requests labeled `patch`,
+the next release should be a patch release (42.3.2).
+If there are neither minor nor patch pull requests, all the others should be labeled `docs` or `pedantic`,
+and you should not make a release.
+
+#### Adding the release to the UPGRADE file
+
+If there are any deprecations, then the release should be minor and the UPGRADE-42.x file should be changed,
+moving the instructions explaining how to bypass the deprecation messages,
+that should hopefully be there, to a new section named `UPGRADE FROM 42.3.1 to 42.4.0`.
+
+#### Compiling the changelog
+
+Each non-pedantic PR should contain a `CHANGELOG` section, that you need to copy manually into the `CHANGELOG.md` file.
+The title is in the following format :
+`[42.3.2](https://github.com/sonata-project/SonataNewsBundle/compare/42.3.1...42.3.2) - YYYY-MM-DD`.
+
+#### Creating the release commit
+
+The changes above should be added to a commit with the following message : 42.3.2
+You should sign this commit with your GPG key, after which all you have to do is push it.
+If you don't have push access, you can still create a PR with the relevant changes
+and have them signed off by someone who has it.
+
+### Major releases
+
+Major releases should be done on a regular basis,
+to avoid branches getting to far of their more stable counterparts:
+the biggest the gap, the harder the merges are.
```

Comment:
IMO here we should explain how many version we support. That will be related with 3 branch management: https://github.com/sonata-project/dev-kit/issues/153

Eg: Releasing a new major 5.0 means that 3.x will be abandoned and 4.x will be legacy

WDYT?